src/core/cgroup.c

```diff
@@ -2074,6 +2076,16 @@ static int unit_update_cgroup(
                 return log_unit_error_errno(u, r, "Failed to create cgroup %s: %m", empty_to_root(u->cgroup_path));
         created = r;
 
+        r = cg_get_path(SYSTEMD_CGROUP_CONTROLLER, u->cgroup_path, NULL, &cgroup_full_path);
```

Comment:
this is a cgroup2 concept.  maybe conditionalize the cg_get_path already under `cg_unified_controller(SYSTEMD_CGROUP_CONTROLLER)`